delegator.py

```diff
@@ -182,7 +182,11 @@ def run(self, block=True, binary=False, cwd=None, env=None):
                 popen_kwargs['cwd'] = cwd
             if env:
                 popen_kwargs['env'].update(env)
-            s = subprocess.Popen(self._popen_args, **popen_kwargs)
+            args = self._popen_args
+            if not isinstance(args, str):
+                # It is recommended to pass a sequence based on docs.
```

Comment:
This comment, although correct, is confusing in this context since we're converting from sequence to string on the next line.

Reply:
I will remove it.